beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deneb.go

```diff
@@ -55,16 +56,25 @@ func (c *blobsBundleCache) prune(minSlot primitives.Slot) {
 	}
 }
 
+type sidecarMetaData struct {
+	Blobs     [][]byte
+	KzgProofs [][]byte
+}
+
 // buildBlobSidecars given a block, builds the blob sidecars for the block.
-func buildBlobSidecars(blk interfaces.SignedBeaconBlock) ([]*ethpb.BlobSidecar, error) {
+func buildBlobSidecars(blk interfaces.SignedBeaconBlock, metadata *sidecarMetaData) ([]*ethpb.BlobSidecar, error) {
```

Comment:
I don't think it adds much to add a `sidecarMetaData` object, given there are only two fields, I think it's cleaner to pass blobs and kzg proofs itself. More readable imo